src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeIndex_Create.cs

```diff
@@ -214,14 +230,69 @@ internal sealed partial class SyntaxTreeIndex
                             containsAwait,
                             containsTupleExpressionOrTupleType),
                     new DeclarationInfo(
-                            declaredSymbolInfos.ToImmutableAndFree()));
+                            declaredSymbolInfos.ToImmutableAndFree()),
+                    new ExtensionMethodInfo(
+                        simpleExtensionMethodInfoBuilder.ToImmutableDictionary(GetKey, GetImmutableValue),
+                        complexExtensionMethodInfoBuilder.ToImmutable()));
             }
             finally
             {
                 Free(ignoreCase, identifiers, escapedIdentifiers);
                 StringLiteralHashSetPool.ClearAndFree(stringLiterals);
                 LongLiteralHashSetPool.ClearAndFree(longLiterals);
+                simpleExtensionMethodInfoBuilder.Free();
+                complexExtensionMethodInfoBuilder.Free();
+            }
+
+            static void AddExtensionMethodInfo(
+                IDeclaredSymbolInfoFactoryService infoFactory,
+                SyntaxNode node,
+                bool containsUsingAliasDirective,
+                int declaredSymbolInfoIndex,
+                DeclaredSymbolInfo declaredSymbolInfo,
+                PooledDictionary<string, ArrayBuilder<int>> simpleInfoBuilder,
+                ArrayBuilder<int> complextInfoBuilder)
+            {
+                if (declaredSymbolInfo.Kind != DeclaredSymbolInfoKind.ExtensionMethod)
+                {
+                    return;
+                }
+
+                // If there's using alias, we don't even try to figure out if the alias is related
+                // to the target type, simply treated it as complex type.
+                if (containsUsingAliasDirective || declaredSymbolInfo.TypeParameterCount > 0)
```

Comment:
i don't think you need to do this.

Reply:
need to document why the TypeParameterCount check is there.  i think i know why, but we shuld doc this.

Reply:
also, do you have tests for this?

Reply:
This check is now removed, since we don't blindly reject generic methods now.